docs/tools/how-to-setup-report-script.md

```diff
@@ -0,0 +1,169 @@
+* running time: how much time this job is in running status, this include retries
+* retries: the retry count of this job
+* status: the status of the job, it could be `WAITING`, `RUNNING`, `SUCCEEDED`, `STOPPED`, `FAILED` and `UNKNOWN`
+* exit code: the exit code of the job, if the job is still in running, it will be value `N/A`
+* cpu allocated: how many vcore allocated to the job, this include the vcore allocated to app master
+* memory allocated: how much memory(GB) allocated to the job, this include the memory allocated to app master
+* max memory usage: maximum memory(GB) usage of this job, it will have value of `N/A` if Pai did not have record of memory usage, maybe due to running time of job is too short or system error
+* gpu allocated: how many gpu card allocated to the job
+
+### gpu
+
+This report is about all gpu util info in cluster.
+
+The report have following columns:
+
+* host_ip: where this gpu installed
+* gpu_id: gpu minor number in the node
+* avg: avg utils during the report time frame
+
+## Prerequisite
+
+You should prepare a node that have access to OpenPai services, the script will need to access hadoop-resource-manager, framework-launcher and Prometheus deployed by OpenPai. This node do not need to have much memory resource and do not need to have GPU cards. You only need to make sure this node will not restart frequently. Usually the master node of the OpenPai cluster is a good choice.
+
+After you choose a node, please make sure you have following software installed:
+
+* python3
+* requests library
+* flask library
+
+If your node is ubuntu node, you can install these software using following commands:
+
+``` sh
+sudo apt-get install -y python3 python3-pip
+pip3 install -r $PAI_DIR/src/tools/reports_requirements.txt
+```
+
+## How to Setup
+
+The [script](../../src/tools/reports.py) has thress actions, `refresh`, `report` and `serve`.
+
+The `refresh` action will tries to collect data from hadoop-resource-manager and framework-launcher, and save the data in sqlite3 DB for future process. The script needs to save data because hadoop-resource-manager will not retain job info too long, if we do not fetch them and save somewhere, we will not be able to generate correct report. We recommend admin run this script every 10 minutes using CRON job.
+
+The `report` action will query data about vc usage and job statistic from sqlite3 DB and generate vc/job/raw_job/gpu csv files, it will also get data from Prometheus to generate alert reports. You can execute this action whenever you want the reports.
+
+The `serve` action will start a http server so outside world can query report through web server instead of using files.
+
+Both `serve` and `report` will need `refresh` being called periodically to fetch data
+from underlaying source.
+
+First, log into the node you choose, put the [script](../../src/tools/reports.py) somewhere, for example, I put it in directory `/home/core/report`, edit the crontab using
+
+``` sh
```

Comment:
Currently it's more like services than tools, why not make it an optional service, like alert-manager.

Reply:
Bin object this, if it is a service, we have to maintain its status, and if it is a tool, the users are responsible for that.

Reply:
I see, but it's not that user friendly, and might mess host envs. Is it necessary to tell user how to uninstall this tool?

Reply:
Yes, maybe we should change it to service in the future.